ex2_system/source/leop/leop.c

```diff
@@ -24,103 +24,95 @@
 #define FOUR_MIN_DELAY pdMS_TO_TICKS(240 * 1000)
 #define TWENTY_SEC_DELAY pdMS_TO_TICKS(20 * 1000)
 
-// LEOP Sequence
+char *deployable_to_str(Deployable_t sw) {
+    switch (sw) {
+    case DFGM:
+        return "DFGM";
+    case UHF_P:
+        return "UHF_P";
+    case UHF_Z:
+        return "UHF_Z";
+    case UHF_S:
+        return "UHF_S";
+    case UHF_N:
+        return "UHF_N";
+    case Port:
+        return "PORT";
+    case Payload:
+        return "PAYLOAD";
+    case Starboard:
+        return "STARBOARD";
+    default:
+        return "Unknown";
+    }
+}
 
 /**
  * @brief
- *      Deploy all deployable systems
- * @return bool
- *      Returns TRUE if all deployables have been deployed
- *      Returns FALSE otherwise
+ *      Deploye specific deployable
+ * @param sw
+ *      Specific Deployable_t to attempt to deploy
+ * @param attempts
+ *      Number of times to retry deploying the Deployable_t
+ * @param deployed_state
+ *      Expected switch status after deployment
  */
-
-bool deploy_all_deployables() {
-    Deployable_t sw;
-
-    int getStatus_retries;
-    uint16_t burnwire_currents[8] = {0};
-
-    for (getStatus_retries = 0; getStatus_retries <= MAX_RETRIES; getStatus_retries++) {
-        sw = DFGM;
-        // Deploy DFGM
-        if ((switchstatus(sw) != 1) && (getStatus_retries != MAX_RETRIES)) {
-            ex2_log("Check #%d: %c not deployed\n", &getStatus_retries, sw);
-            ex2_log("Activated %c\n", sw);
-            activate(sw);
-            vTaskDelay(TWENTY_SEC_DELAY);
-        } else if ((switchstatus(sw) != 1) && (getStatus_retries == MAX_RETRIES)) {
-            ex2_log("Check #%d: %c not deployed, exiting the LEOP sequence.\n", &getStatus_retries, sw);
-            return false;
-        }
+void deploy(Deployable_t sw, int attempts, int deployed_state) {
+    for (int deployment_attempt = 0; deployment_attempt < attempts; deployment_attempt++) {
+        sys_log(INFO, "LEOP: Starting burn %d for %s\n", deployment_attempt, deployable_to_str(sw));
+        activate(sw);
```

Comment:
Please add a 20 second delay between attempts. Otherwise, this will be equivelant to running one looooong deployment because the resistor doesn't cool down. This could damage the resistor because it would reach a much higher temperature than necessary.

Reply:
https://github.com/AlbertaSat/ex2_obc_software/pull/239/commits/61d132fc008342fc6c39efc85f935c4d91ca645a